kb/2025-05-26/velero-backup-restore.md

```diff
@@ -0,0 +1,358 @@
+
+Confirm that Velero is installed and running:
+
+```sh
+kubectl -n velero get po
+```
+
+```sh
+NAME                      READY   STATUS    RESTARTS         AGE
+node-agent-875mr          1/1     Running   0                1d
+velero-745645565f-5dqgr   1/1     Running   0                1d
+```
+
+Configure the `velero` CLI to output the backup and restore status of CSI objects:
+
+```sh
+velero client config set features=EnableCSI
+```
+
+## Deploy the NFS CSI and Example Server
+
+Follow the instructions in the [NFS CSI documentation](https://github.com/kubernetes-csi/csi-driver-nfs/blob/master/deploy/example/README.md) to set up the NFS CSI driver, its storage class, and an example NFS server.
+
+The NFS CSI volume snapshotting capability must also be enabled following the instructions [here](https://github.com/kubernetes-csi/csi-driver-nfs/tree/master/deploy/example/snapshot).
+
+Confirm that the NFS CSI and example server are running:
+
+```sh
+kubectl get po -A -l 'app in (csi-nfs-node,csi-nfs-controller,nfs-server)'
+```
+
+```sh
+NAMESPACE     NAME                                  READY   STATUS    RESTARTS    AGE
+default       nfs-server-b767db8c8-9ltt4            1/1     Running   0           1d
+kube-system   csi-nfs-controller-5bf646f7cc-6vfxn   5/5     Running   0           1d
+kube-system   csi-nfs-node-9z6pt                    3/3     Running   0           1d
+```
+
+The default NFS CSI storage class is named `nfs-csi`:
+
+```sh
+kubectl get sc nfs-csi
+```
+
+```sh
+NAME      PROVISIONER      RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
+nfs-csi   nfs.csi.k8s.io   Delete          Immediate           true                   14d
+```
+
+Similarly, the default NFS CSI volume snapshot class is named `csi-nfs-snapclass`:
+
+```sh
+kubectl get volumesnapshotclass csi-nfs-snapclass
+```
+
+```sh
+NAME                DRIVER           DELETIONPOLICY   AGE
+csi-nfs-snapclass   nfs.csi.k8s.io   Delete           14d
+```
+
+## Preparing the Virtual Machine and Image
+
+Refer to the Harvester documentation on how to perform the following steps using the NFS CSI storage class:
+
+ * [Create a virtual machine image](https://docs.harvesterhci.io/v1.6/advanced/csidriver#virtual-machine-image-creation)
+ * [Create a virtual machine](https://docs.harvesterhci.io/v1.6/advanced/csidriver#virtual-machine-creation)
+
+This example uses the [Ubuntu 24.04 raw image](https://cloud-images.ubuntu.com/minimal/releases/noble/) to create a virtual machine with NFS root and data volumes in the custom `demo-src` namespace:
+
+![image](vm-create.png)
+
+:::note
+
+For CSI snapshotting to work, the NFS volumes must have the `volumeMode` set to `Filesystem`.
+
+:::
+
+Access the virtual machine via SSH, and add some files to both the root and data volumes.
+
+The data volume needs to be partitioned, with a file system created and mounted before files can be written to it.
```

Comment:
Is this true if the `volumeMode` set to `Filesystem` as stated in the note above?

Reply:
Correct, because the VM will still see it as a disk, unlike a shared file system over `virtiofs`.